web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php

```diff
@@ -427,4 +428,14 @@ public static function ensureUrlIsString(string|GeneratedUrl $url): string {
     return $url;
   }
 
+  /**
+   * Get the instant loan configuration.
+   *
+   * @return mixed[]
+   *   The instant loan configuration.
+   */
+  public static function getInstantLoanConfig() {
+    return \Drupal::configFactory()->get('dpl_instant_loan.settings')->get() ?? [];
```

Comment:
The approach to using configuration here is different compared to the one we use for branches.

Other settings are wrapped in services and injected into the controller. The controller then again has methods to convert the settings to JSON as expected by the apps. The service also knows how to use the setting to control caches.

This setting is retrieved directly from a static method and then passed to the app. It does not deal with caching when the configuration changes.

Personally I appreciate the type handling for the approach used with branch settings I recognize that it may be over-complicated. [There are already other settings where it has not been implemented](https://github.com/reload/dpl-cms/blob/main/web/modules/custom/dpl_library_agency/src/Form/GeneralSettingsForm.php#L210-L210), [Drupal configuration objects can also be used with cache handling on their own](https://github.com/drupal/drupal/blob/f965e40015127c57b48c853bd60bb81a7d5cd3bb/core/lib/Drupal/Core/Config/ConfigBase.php#L27) and [serialization to JSON is also handled elsewhere](https://github.com/reload/dpl-cms/blob/main/web/modules/custom/dpl_react/dpl_react.module#L129-L129).

With all this in mind I think there are at least some changes that should be made to the current implementation:

1. Use `$this->config('dpl_instant_loan.settings')` as provided by the controller class to retrieve the config in the controller instead of using `\Drupal::configFactory()`
2. Add the configuration object as a cachable dependency to the render array for the material app
3. Consider whether it is possible to document the shape of the instant loans config expected by the React application. I do not have any good suggestion. One option would be to add it to the `getInstantLoanConfig()` docblock but if we need a `Config` object for cache management then we will need to change the signature of this function anyway.

Reply:
Added e config service 👍